ftplugin/php/commands.vim

```diff
@@ -1,130 +1,101 @@
 
 ""
 " Goto the definition of the class, method or function under the cursor. Open
 " the definition in the current window.
-command! -nargs=0 PhpactorGotoDefinition call phpactor#GotoDefinition()
+command! -buffer -nargs=0 PhpactorGotoDefinition call phpactor#GotoDefinition()
 
 ""
 " Goto type (class) of the symbol under the cursor.
-command! -nargs=0 PhpactorGotoType call phpactor#GotoType()
+command! -buffer -nargs=0 PhpactorGotoType call phpactor#GotoType()
 
 ""
 " As with @command(PhpactorGotoDefinition) but open in a vertical split.
-command! -nargs=0 PhpactorGotoDefinitionVsplit call phpactor#GotoDefinitionVsplit()
+command! -buffer -nargs=0 PhpactorGotoDefinitionVsplit call phpactor#GotoDefinitionVsplit()
 
 ""
 " As with @command(PhpactorGotoDefinition) but open in a horizontal split.
-command! -nargs=0 PhpactorGotoDefinitionHsplit call phpactor#GotoDefinitionHsplit()
+command! -buffer -nargs=0 PhpactorGotoDefinitionHsplit call phpactor#GotoDefinitionHsplit()
 
 ""
 " As with @command(PhpactorGotoDefinition) but open in a new tab
-command! -nargs=0 PhpactorGotoDefinitionTab call phpactor#GotoDefinitionTab()
+command! -buffer -nargs=0 PhpactorGotoDefinitionTab call phpactor#GotoDefinitionTab()
 
 ""
 " Load all implementations of the class under the cursor into the quick-fix
 " list.
-command! -nargs=0 PhpactorGotoImplementations call phpactor#GotoImplementations()
+command! -buffer -nargs=0 PhpactorGotoImplementations call phpactor#GotoImplementations()
 
 ""
 " Show information about the symbol under the cursor.
-command! -nargs=0 PhpactorHover call phpactor#Hover()
+command! -buffer -nargs=0 PhpactorHover call phpactor#Hover()
 
 ""
 " Show the context menu for the current cursor position.
-command! -nargs=0 PhpactorContextMenu call phpactor#ContextMenu()
+command! -buffer -nargs=0 PhpactorContextMenu call phpactor#ContextMenu()
 
 ""
 " Copy the current file - updating the namespace and class name according to
 " the new file location and name
-command! -nargs=0 PhpactorCopyFile call phpactor#CopyFile()
+command! -buffer -nargs=0 PhpactorCopyFile call phpactor#CopyFile()
 
 ""
 " Move the current file - updating the namespace and class name according to
 " the new file location and name
-command! -nargs=0 PhpactorMoveFile call phpactor#MoveFile()
-
-""
-" List all installed extensions
-command! -nargs=0 PhpactorExtensionList call phpactor#ExtensionList()
-
-""
-" Install an extension
-command! -nargs=1 PhpactorExtensionInstall call phpactor#ExtensionInstall(<q-args>)
-
-""
-" Remove an extension
-command! -nargs=1 PhpactorExtensionRemove call phpactor#ExtensionRemove(<q-args>)
+command! -buffer -nargs=0 PhpactorMoveFile call phpactor#MoveFile()
 
 ""
 " Create a new class. You will be offered a choice of templates.
-command! -nargs=0 PhpactorClassNew call phpactor#ClassNew()
+command! -buffer -nargs=0 PhpactorClassNew call phpactor#ClassNew()
```

Comment:
Assuming we have implemented the idea of project root per buffer and that we can manually assign some project to non-php buffer (e.g. composer.json in a newly created project), we could move this command to `plugin`.

Reply:
Incidentally - there is also a case for using f.e. `PhpactorGotoDefinition` (and similar) in `.yml`, `.xml`, Gherkin files, etc.

Reply:
We probably need to decide which commands are global, and which only apply to buffers (or make them all available unconditionally)

Reply:
I went pretty quickly other the commands so I might have missed some for sure.
For the go to definition I didn't realized we could use them in non PHP files...
For the new class it's something that might be appreciated anywhere indeed no matter if the file is part of a project or not.

To be honest I never use the commands, go to definition is mapped on `<C-]>` and `<C-w><C-]>` and I don't use the new class functionality because I have snippets that can do that for me a lot better.
Just to say that I might not have a realistic vision of when to use them.

Reply:
> For the new class it's something that might be appreciated anywhere indeed no matter if the file is part of a project or not.

Right, although providing the project root to this command (or use the assigned project root to the current buffer previously) would help resolve namespace (or file path if there would be possible to adding it given a namespace) and generate suggestions.

Reply:
I think this should be handle in `phpactor#newClass()` directly.
We can easly retrieve the current buffer and find out if it belongs to a project or not.

Reply:
It would work if the current buffer belongs to any project. If the buffer is for example empty, special or help buffer providing the root can be still needed.

Reply:
So what about this list ?
I would like to be able to merge this so we can move on.

For `PhpactClassNew` I'm not against it, but I don't really have an opinion since I never use It.
For the navigation commands (goto and all that) I think it's better to declare them in their own `filetype/cucumber.vim` file.
Especially because this support came from an extension that is not bundled with phpactor by default.
We could even say that the extension should do it itself, but we don't have such a feature right now but we should separate things clearly.

Reply:
Independently of the place of the command definition I can assign a mapping or any custom command to `phpactor#ClassNew()` directly.